core/domain/exp_domain.py

```diff
@@ -3338,7 +3338,7 @@ def _choices_should_be_unique_and_non_empty(
         if state_dict['interaction']['solution'] is not None:
             solution = state_dict['interaction']['solution']['correct_answer']
             if isinstance(solution, list) and any(
-                invalid_choice['html'] in solution for invalid_choice in
+                invalid_choice['content_id'] in solution for invalid_choice in
```

Comment:
Why do we change this?

Reply:
Hm, I think I don't have enough context to properly evaluate this. If @DubeySandeep can approve this, I'm happy to approve it as well.

Reply:
This looks good to me, thanks for adding it @lkbhitesh07 !

I just have two things remaining to ensure:

1. Validate state contains solutions only if interaction.can_have_solution.
2. Validate all the keys are present in the customization args. (Not sure whether it already exists.)

These two validations are not a blocker from my side but just ensure with your mentor and LaCE team for the requirements. (Just trying to avoid future migration for similar stuff, if it's really needed!)

Reply:
Hmm, Well these were not initially included in the project and we did not had any discussions regarding this. For the `2` one I particularly do not remove any keys from the customization args AFAIK.

Hey @vojtechjelinekm Just wanted to ping you here for this particular discussion.
Thanks

Reply:
Let's not add any more validations at this stage, we can do them later.